tests/subsys/usb/desc_sections/src/desc_sections.c

```diff
@@ -204,35 +204,35 @@ static void test_desc_sections(void)
 
 	TC_PRINT("__usb_descriptor_start %p\n", __usb_descriptor_start);
 	TC_PRINT("__usb_descriptor_end %p\n",  __usb_descriptor_end);
-	TC_PRINT("USB Descriptor table size %d\n",
-		 (int)__usb_descriptor_end - (int)__usb_descriptor_start);
+	TC_PRINT("USB Descriptor table size %ld\n",
+		 (intptr_t)__usb_descriptor_end - (intptr_t)__usb_descriptor_start);
```

Comment:
Maybe a silly comment, but:
If this change is just for printing pointers arithmetic results, Would it be a cleaner to cast them directly to long or int here? i.e.:
```
TC_PRINT("USB Descriptor table size %ld\n",
		 (long)((intptr_t)__usb_descriptor_end - (intptr_t)__usb_descriptor_start));
```
For any pointer and long size that would be correct and warning free, or?

Or did you have some other reason for the change in posix_cheats?